src/Native/Runtime/StackFrameIterator.cpp

```diff
@@ -1739,7 +1740,7 @@ bool StackFrameIterator::GetHijackedReturnValueLocation(PTR_RtuObjectRef * pLoca
 
 void StackFrameIterator::SetControlPC(PTR_VOID controlPC)
 {
-    m_OriginalControlPC = m_ControlPC = controlPC;
+    m_OriginalControlPC = m_ControlPC = dac_cast<PTR_VOID>(PCODEToPINSTR(dac_cast<PCODE>(controlPC)));
```

Comment:
It works on Linux; function pointers in the binary are even numbers, and the discrepancy arises when libunwind returns odd addresses. Should this be under `#ifdef PLATFORM_UNIX`?

Reply:
Do you mean that the return addresses like `PointerToRhpRethrow2` do not have the thumbbit set on Unix?

If it is the case, I think it would be best to set the bit for them in the `EQUALS_RETURN_ADDRESS` macro. Something like: 

```
#if defined(PLATFORM_UNIX) && defined(_TARGET_ARM_)
// The captured return addresses do not have the thumb bit set on Unix
#define EQUALS_RETURN_ADDRESS(x, func_name) (dac_cast<TADDR>(x) == PINSTRToPCODE(dac_cast<TADDR>(PointerTo ## func_name)))
#else
... the existing code
#endif
```

Reply:
Hm, I suppose it's also possible to make the `EXPORT_POINTER_TO_ADDRESS` unix/arm version of the macro add 1 to the address and avoid `ifdef`s like so:

```
.macro EXPORT_POINTER_TO_ADDRESS Name

1:

        .rodata
        .align      4
C_FUNC(\Name):
        .word       1b + 1                      // add 1 to indicate thumb code
        .global     C_FUNC(\Name)
        .text

.endm
```

Either way, if we go this route, we'd need to return this back because `pMethodStartAddress` is also even: https://github.com/dotnet/corert/blob/5f4dd60e6130127e2c5871c2d81723e519097cfe/src/Native/Runtime/unix/UnixNativeCodeManager.cpp#L149

Reply:
> make the `EXPORT_POINTER_TO_ADDRESS` unix/arm version of the macro add 1 to the address 

Yes, that would work too - if the ARM assembler accepts it.

> if we go this route, we'd need to return this back because pMethodStartAddress is also even

I assume that this would be just in `UnixNativeCodeManager.cpp`. That looks fine to me.

Reply:
Ok, I made the changes